libr/parse/p/parse_tms320_pseudo.c

```diff
@@ -0,0 +1,144 @@
+/* radare - LGPL - Copyright 2020 - pancake */
+
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+
+#include <r_lib.h>
+#include <r_flag.h>
+#include <r_anal.h>
+#include <r_parse.h>
+
+static int replace(int argc, const char *argv[], char *newstr) {
+	int i, j, k;
+	struct {
+		int narg;
+		char *op;
+		char *str;
+	} ops[] = {
+		{2, "mov", "1 = 2"},
+		{3, "band", "1 = 2 & 3"},
+		{0, "reti", "ret"},
+		{2, "invalid", "?"},
+		{0, NULL}};
+	if (!newstr) {
+		return false;
+	}
+
+	for (i = 0; ops[i].op; i++) {
+		if (ops[i].narg) {
+			if (argc - 1 != ops[i].narg) {
+				continue;
+			}
+		}
+		if (!strcmp(ops[i].op, argv[0])) {
+			for (j = k = 0; ops[i].str[j]; j++, k++) {
+				if (IS_DIGIT (ops[i].str[j])) {
+					const char *w = argv[ops[i].str[j] - '0'];
+					if (w) {
+						strcpy (newstr + k, w);
+						k += strlen (w) - 1;
+					}
+				} else {
+					newstr[k] = ops[i].str[j];
+				}
+			}
+			newstr[k] = '\0';
+			if (argc == 4 && argv[2][0] == '[') {
+				strcat (newstr + k, "+");
+				strcat (newstr + k + 3, argv[2]);
+			}
+			return true;
+		}
+	}
+
+	/* TODO: this is slow */
+	newstr[0] = '\0';
+	for (i = 0; i < argc; i++) {
+		strcat (newstr, argv[i]);
+		strcat (newstr, (i == 0 || i == argc - 1) ? " " : ",");
+	}
+	return false;
+}
+
+static int parse(RParse *p, const char *data, char *str) {
+	if (!strncmp (data, "|| ", 2)) {
+		data += 3;
+	}
+	char w0[256], w1[256], w2[256], w3[256];
+	int i, len = strlen (data);
+	char *buf, *ptr, *optr;
+
+	if (len >= sizeof (w0)) {
+		return false;
+	}
+	// malloc can be slow here :?
+	if (!(buf = malloc (len + 1))) {
+		return false;
+	}
+	memcpy (buf, data, len + 1);
+
+	if (*buf) {
+		r_str_replace_char (buf, '(', ' ');
+		r_str_replace_char (buf, ')', ' ');
+		*w0 = *w1 = *w2 = *w3 = '\0';
+		ptr = strchr (buf, ' ');
+		if (!ptr) {
+			ptr = strchr (buf, '\t');
+		}
+		if (ptr) {
```

Comment:
can we have a small comment before this big chunk of code that explains a little bit how the text is supposed to be transformed? Something like:
```
// Convert "mov rax, [rip + 123213]" into "mov rax, mysuperflag"
```